crypto/sig/src/lib.rs

```diff
@@ -25,16 +24,12 @@ pub use schnorrkel::{Signature, SignatureError, SIGNATURE_LENGTH};
 /// Returns:
 /// * A 64-byte Schnorrkel Signature object which can be converted to and from bytes using its API.
 pub fn sign(context_tag: &[u8], private_key: &RistrettoPrivate, message: &[u8]) -> Signature {
-    // Nonce is hash( private_key || message )
-    // FIXME: We should probably hash the context_tag in as well.
-    // Or just use something like merlin instead of Blake2b256.
-    // In that case we make the assumption that Keccak, which underlies STROBE,
-    // is a strong Pseudorandom permutation, and that consequently Merlin with
-    // a partially secret input is a PRF.
-    let mut hasher = Blake2b256::new();
-    hasher.input(private_key.to_bytes());
-    hasher.input(message);
-    let nonce = hasher.result();
+    let mut transcript = Transcript::new(b"SigningNonce");
+    transcript.append_message(b"context", &context_tag);
+    transcript.append_message(b"private", &private_key.to_bytes());
+    transcript.append_message(b"message", &message);
+    let mut nonce = [0u8; 32];
+    transcript.challenge_bytes(b"nonce", &mut nonce);
```

Comment:
it might be a little easier to read if we scoped this transcript, like

```
let nonce = {
    let mut transcript = Transcript::new(b"SigningNonce");
    ...
    let mut nonce = [0u8;32];
    transcript.challenge_bytes(b"nonce", &mut nonce);
};
```
just so that there's no confusion between this transcript and the later transcript?

Reply:
👍